Sources/HAP/Controllers/PairSetupController.swift

```diff
@@ -32,24 +38,56 @@ class PairSetupController {
             throw Error.invalidPairingMethod
         }
 
+        // If the accessory is already paired it must respond with
+        // Error_Unavailable
+        if device.state == .paired {
+            return [
+                (.state, Data(bytes: [PairSetupStep.startResponse.rawValue])),
+                (.error, Data(bytes: [PairError.unavailable.rawValue]))
+            ]
+        }
+
+        // If the accessory has received more than 100 unsuccessful
+        // authentication attempts it must respond with
+        // Error_MaxTries
+        // TODO
+
+        // If the accessory is currently performing a Pair Setup operation with
+        // a different controller it must respond with
+        // Error_Busy
+        if device.state == .pairing {
+            return [
+                (.state, Data(bytes: [PairSetupStep.startResponse.rawValue])),
+                (.error, Data(bytes: [PairError.busy.rawValue]))
+            ]
+        }
+
+        // Notify listeners of the pairing event and record the paring state
+        device.changePairingState(.pairing)
```

Comment:
We don't reset the state after some timeout. This might leave the device in the pairing state, unable to be paired after some failure. I will have a look at how the controller can be decoupled from the device, and maybe rely on the session state that is being passed in. I will get back on this later.

Reply:
OK let me know if you need me to look at anything